diag/diagnostics.go

```diff
@@ -70,3 +70,55 @@ func (diags Diagnostics) HasError() bool {
 
 	return false
 }
+
+// ErrorsCount returns the number of Diagnostic in Diagnostics that are SeverityError.
+func (diags Diagnostics) ErrorsCount() int {
+	count := 0
+
+	for _, d := range diags {
+		if SeverityError == d.Severity() {
+			count++
+		}
+	}
+
+	return count
+}
+
+// WarningsCount returns the number of Diagnostic in Diagnostics that are SeverityWarning.
+func (diags Diagnostics) WarningsCount() int {
```

Comment:
Nit: both those `*Count` methods could be implemented as a `len(Call_to_non_Count_methods)`.

Reply:
Good point. Fixed.